core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,26 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Test that unknown field trigger ParsingException.
+     * To find the right position in the root query, we add a marker as `queryName` which
+     * all query builders support. The added bogus field after that should trigger the exception.
+     * Queries that allow arbitrary field names at this level need to override this test.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        assumeFalse("Skip test in unlikely event that we have _marker_ in some random string", testQuery.toString().contains("_marker_"));
+        testQuery.queryName("#marker#"); // to find root query to add additional bogus field there
+        String queryAsString = testQuery.toString().replace("\"#marker#\"", "\"#marker#\", \"bogusField\" : \"someValue\"");
```

Comment:
sneaky! :) this works better than my previous suggestion, definitely. Still, should we just double check (sorry I am paranoid) that there's one and only one occurrence of the marker in the whole query? If there's none we should error, if there's more we should skip the test?